src/Nethermind/Nethermind.Blockchain.Test/Validators/TxValidatorTests.cs

```diff
@@ -171,7 +171,6 @@ public bool Before_eip_2930_has_to_be_legacy_tx(TxType txType, bool eip2930)
     [TestCase(TxType.AccessList, true, false, ExpectedResult = true)]
     [TestCase(TxType.EIP1559, true, false, ExpectedResult = false)]
     [TestCase(TxType.EIP1559, true, true, ExpectedResult = true)]
-    [TestCase((TxType)100, true, false, ExpectedResult = false)]
```

Comment:
The only test case removed since it's nonsensical: there is no know `TxType` with code `100`

Reply:
Well the point of the this test case was that it failed?

Reply:
That case would result in the `TxValidator` rejecting the transaction as malformed. Now, we fail earlier in the decoding stage (required when computing the transaction hash).

The difference now is that we throw an exception instead of returning `false`.

Reply:
do not assume the internal code when writing tests, tests are idnependent

Reply:
throwing exception instead of returning false is a potentially big change in many places in the code and may affect entire p2p and evm processing behaviour

significant amount of testing would have to be added

Reply:
What do you think @LukaszRozmej ? When we discussed this refactor we mentioned that on unknown transaction types we would throw. Should we move forward with that approach or do you want to go back to the drawing board?

Reply:
> Would be good to write a test for `NewPayload` with unknown tx type

Could you point me to where similar tests (if any) exists? I'm a bit lost on what we're trying to test.

Reply:
We need to guarantee that NewPayload returns correct error, we can add it in `EngineModuleTests` class.

Reply:
Added test and fixed exceptions, same error now on master and this branch